tests/wrapper/test_simulation_object_updating_component.py

```diff
@@ -110,3 +110,15 @@ def test_node_refs(self, configured_component: SimulationObjectUpdatingComponent
                     )
 
                     assert node in (edge.to_node, edge.from_node)
+
+    def test_stale_train_is_removed(
+        self,
+        configured_component: SimulationObjectUpdatingComponent,
+        train: Train,
+        results,
+        all_trains,
+    ):
+        configured_component.next_tick(1)
+        configured_component.next_tick(2)
+
+        assert len(configured_component.trains) == 0
```

Comment:
Probably I just did not understand your code fully. But I could not find where the fixture `configured_component` got any trains added. So are you really testing the deletion of trains here?

Reply:
You are correct, fixed that :)